entities/schema/schema.go

```diff
@@ -17,6 +17,12 @@ import (
 	"github.com/weaviate/weaviate/entities/models"
 )
 
+// Finder direct access to get the class without quiring the whole schema.
+type Finder interface {
```

Comment:
I can remove it ,no problem, it's just to use it as a common instead of dupp

Reply:
actually it's needed see [here](https://github.com/weaviate/weaviate/pull/4016/files#diff-3879dfc1c5c38076c6421841c9e45dbd9dc111da6d14665abec66b667d6a244eR30) because we need away  to inject the `ReadOnlyClass`

Reply:
I keep the interface simple by removing this. I don't see the duplication argument, as interfaces depend on the consumer and the context in which they are used hence duplication might make sense here.

Reply:
it's also because of the dependnacy injection,  for example [here](https://github.com/weaviate/weaviate/pull/4016/files#diff-9f2c94bdc64e7b6ba3140256fc1b16171f8f1fb4cd3e9f2ce2a3bb119d925441R228)  converting `FindPropertyDataTypeWithRefs`  to function require that

Reply:
A simple approach is for the class to be provided by the caller. This way, we avoid introducing dependencies with might be be necessary in the first place

Reply:
the class is needed under specific situations, it would be unnecessary to get the class from the caller  [w.r.t](https://github.com/weaviate/weaviate/pull/4016/files#diff-9f2c94bdc64e7b6ba3140256fc1b16171f8f1fb4cd3e9f2ce2a3bb119d925441R267)

Reply:
removed